app/scripts/lib/local-store.js

```diff
@@ -11,6 +12,38 @@ export default class ExtensionStore {
     if (!this.isSupported) {
       log.error('Storage local API not available.');
     }
+    this.dataPersistenceFailing = false;
+  }
+
+  async init() {
+    this.initialStoreState = await this.get();
```

Comment:
Are we sure this is only run after the migrator? The migrator is what sets the version if the extension has been updated.

Reply:
It seems to be, but to be sure I could move it right after this: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/background.js#L295

Reply:
wait that won't work...

Reply:
in my testing it does a appear to be called after the migrator has run, but we could, alternatively, call `init()` [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/background.js#L295), pass down the same instantiation of `localStore` from `background.js` through `setupController` in the `MetamaskController` constructor and then into `createMetaRPCHandler` so that it accesses the same `initialStoreState` class variable when run in the RPC client

Reply:
Yeah I was just thinking the same thing!

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/fdeb4399ca7665ee03bb979fceb5ec6c07abe5f2)